src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -817,15 +817,15 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
             }
         }
 
-        // Ensure the `IsNotPromotable` qualification is preserved.
+        // Ensure the `IsNotConst` qualification is preserved.
         // NOTE(eddyb) this is actually unnecessary right now, as
         // we never replace the local's qualif, but we might in
         // the future, and so it serves to catch changes that unset
         // important bits (in which case, asserting `contains` could
         // be replaced with calling `insert` to re-set the bit).
         if kind == LocalKind::Temp {
```

Comment:
Again, here, this seems risky, why don't we do this for all locals?
I think relying on the `Var` / `Temp` distinction was a bit of a mistake... I mean, here, as opposed to in the analysis in `promote_consts`.